packages/dds/merge-tree/src/mergeTree.ts

```diff
@@ -1971,50 +2064,52 @@ export class MergeTree {
 		const localOverlapWithRefs: ISegmentLeaf[] = [];
 		const removedSegments: SegmentWithInfo<IHasRemovalInfo, ISegmentLeaf>[] = [];
 
+		const createRefFromSequencePlace = (
+			place: InteriorSequencePlace,
+		): LocalReferencePosition => {
+			const { segment: placeSeg, offset: placeOffset } = this.getContainingSegment(
+				place.pos,
+				perspective,
+			);
+			assert(
+				isSegmentLeaf(placeSeg) && placeOffset !== undefined,
+				0xa3f /* segments cannot be undefined */,
+			);
+			return this.createLocalReferencePosition(
+				placeSeg,
+				placeOffset,
+				ReferenceType.StayOnRemove,
+				{
+					side: place.side,
```

Comment:
can this just go inside the obliterate info, rather than in the flat properties of the ref. it make the code easier to follow and debug if we can keep all the obliterate stuff together